Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -47,43 +47,50 @@ public final class PrometheusCollectorRegistry: Sendable {
         }
     }
 
-    private struct MetricWithHelp<Metric: AnyObject & Sendable>: Sendable {
-        var metric: Metric
-        let help: String
-    }
-
     private enum HistogramBuckets: Sendable, Hashable {
         case duration([Duration])
         case value([Double])
     }
 
-    /// A collection of metrics, each with a unique label set, that share the same metric name.
-    /// Distinct help strings for the same metric name are permitted, but Prometheus retains only the
-    /// first one. For an unlabelled metric, the empty label set is used as the key, and the
-    /// collection contains only one entry. Finally, for clarification, the same metric name can
-    /// simultaneously be labeled and unlabeled.
-    /// For histograms, the buckets are immutable for a MetricGroup once initialized with the first
+    /// A MetricFamily.
+    ///
+    /// A metric name can either map to multiple labeled metrics OR a single unlabeled metric,
```

Comment:
Since this is a strict requirement we should make sure it can't be violated.

I'd make the metricsByLabelSets a `let`, or make it `private`, and make a new method through which we modify or make a new family instance; there we'd centralize all the checks.

You could also do a small enum for the two cases if you want inside of the MetricFamily `enum Metrics { case labelled([LabelsKey: Metric]); case unlabelled(Metric) }` which will help to avoid violating the rule 👀

Reply:
True 😉 pushed a suggestion that hopefully addresses all comments in this regard!